### DISCUSSION TITLE
Adding a project reference to StereoKitC & StereoKitC_UWP

### DISCUSSION THREAD

**Discussion by austinbhale** [StereoKit/StereoKit#569] [Q&A] [answered]
Heyo, I'd like to streamline the process of adding StereoKit projects to my solution and the only way to get the StereoKitC projects to build is by replacing `$(SolutionDir)` with `$(ProjectDir)..\` in their respective project files. For example, this will look for your Tools folder in my solution directory instead of StereoKit's: https://github.com/StereoKit/StereoKit/blob/f4f3df650dcc99582441dc9cd73789e45c0d4673/StereoKitC/StereoKitC.vcxproj#L112 I looked into MSBuild about creating Directory.Build.props/.targets files to potentially override `SolutionDir` for this project, but couldn't get it to work. Do you plan on keeping references to SolutionDir?
Thanks!

**Comment by maluoi** [comment, marked as answer]
Oh that's kinda odd! Are you opening the `StereoKitC.vcxproj` by itself, or using the `StereoKit.sln` in the root directory? `$(SolutionDir)` _should_ work fine from there. And if you're using the vcxproj, is there a particular reason for it?

**Reply by austinbhale** [reply]
I’m adding the StereoKitC, StereoKit, and test projects as existing project references to a separate solution. I find it's easier to debug SK code when all our projects are kept in the same solution and added as references. Though, if I make an edit to a StereoKitC source file from my solution, then I’d have to build the project inside StereoKit's solution, instead of directly in my solution, since the $(SolutionDir) output is the current directory of the solution it's in.

With a Directory.Solution.props file for MSBuild, you can override the SolutionDir value when building SK. But it seems VS probably won't support it: https://github.com/dotnet/msbuild/issues/8038#issuecomment-1283550169. Which means we couldn't build these SK projects in VS from our solution without some MSBuild command or alternative.

Essentially, I’m looking for the best way to debug SK managed & native code via another VS solution, where I find individual project references v helpful

**Reply by austinbhale** [reply]
One idea: what if every $(SolutionDir) is replaced by something like $(StereoKitSolutionDir) or % StereoKitSolutionDir%, then in the StereoKit solution there'd be a global constant StereoKitSolutionDir=$SolutionDir? Then any other solution could specify the directory StereoKit is in. I haven't tried this ~hacky idea out yet, but I’ll update this thread once I do

**Reply by maluoi** [reply]
I think your `$(ProjectDir)..\` idea is actually pretty solid! I haven't been referencing the SK projects like you are, so I've just not run into this before.

I don't believe there's any fundamental thing preventing using that either? I just at one point felt it was cleaner to have more absolute file names, and switched everything to `$(SolutionDir)`. Your way feels a bit more... portable!

I'll switch this over shortly :)

**Reply by austinbhale** [reply]
Awesome, thank you so much :) portable, indeed!

**Reply by maluoi** [reply]
I added this (5048180ee93ddcce1d4ffc5768e74de63e3a56d7)! Seems to work for the projects I have, but I'm not stretching the edges. There's a few that I felt a little uncertain about, but do let me know if it works for you, or if there's still some changes needed!

**Reply by austinbhale** [reply]
Yes, that works! The only one I couldn't build from another solution was the [StereoKitTest_Maui](https://github.com/StereoKit/StereoKit/blob/5048180ee93ddcce1d4ffc5768e74de63e3a56d7/Examples/StereoKitTest/StereoKitTest_Maui/StereoKitTest_Maui.csproj#L55) project but that's no biggie. Thanks for this - building SK from source feels even more of a breeze now!
